pkg/requester/ranking/features.go

```diff
@@ -74,15 +74,14 @@ func (s *featureNodeRanker[Key]) rankNode(ctx context.Context, node model.NodeIn
 			}
 		}
 
-		log.Ctx(ctx).Trace().Stringer("Requirement", requiredKey).Bool("Supported", found).Send()
 		if !found {
 			// Target wasn't found – we can end early as we won't use this node.
-			return requester.RankUnsuitable
+			return requester.RankUnsuitable, fmt.Sprintf("does not support %T %s, only %s", requiredKey, requiredKey, providedKeys)
 		}
 	}
 
 	// Node provides all the specified required types.
-	return requester.RankPreferred
+	return requester.RankPreferred, "provides all the specified required types"
```

Comment:
I figured it was at worst just more documentation and I would rather return a meaningful string rather than an empty one. In the future it may be helpful if diagnosing why some nodes are preferred over others.